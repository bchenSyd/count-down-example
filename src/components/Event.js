import React, { Component } from "react";
import { diffTimeInSeconds, displayTimeDiff } from "../common/timeUtils";


class Event extends Component {
  state = {
    diffInSeconds: 0
  };

  interval = undefined;

  componentWillMount() {
    const {
      event: { AdvertisedStartTime: startTime }
    } = this.props;
    this.setState({
      diffInSeconds: diffTimeInSeconds(startTime)
    });
  }

  componentWillReceiveProps(nextProps) {
    const {
      event: { AdvertisedStartTime: startTime }
    } = nextProps;

    this.setState({
      diffInSeconds: diffTimeInSeconds(startTime)
    });
  }

  componentDidMount() {
    const self = this;
    this.interval = setInterval(() => {
      const { diffInSeconds } = self.state;
      self.setState({
        diffInSeconds: diffInSeconds - 1
      });
    }, 1000);
  }

  componentWillUnmount() {
    if (this.interval) {
      clearInterval(this.interval);
    }
  }

  render() {
    const { event } = this.props;
    const { diffInSeconds } = this.state;
    const {
      EventName,
      Venue: { Venue }
    } = event;

    return (
      <div className="event">
        <div className="count-down">{displayTimeDiff(diffInSeconds)}</div>
        <div>
          <div className="event-name">{EventName}</div>
          <div>{Venue}</div>
        </div>
      </div>
    );
  }
}

export default Event;
