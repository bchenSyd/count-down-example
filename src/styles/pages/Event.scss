.Event {
  display: flex;
  flex-direction: column;
  justify-content: center;
  background-color: #fff;
  border-radius: 4px;
  margin: 10px;
  padding: 6px;
  height: 140px;
  width: 150px;
  .count-down {
    font-size: 1.5em;
    color: #ff4b3d;
    margin-bottom: 20px;
  }
  .event-name {
    font-size: 0.6em;
  }
}
