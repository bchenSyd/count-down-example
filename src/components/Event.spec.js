import React from "react";
import renderer from "react-test-renderer";
import Event from "./Event";

jest.useFakeTimers();
describe("Event", () => {
  it("should render a event tile properly", () => {
    const event = {
      EventID: 30717084,
      EventName: "Race 3",
      EventTypeDesc: "Greyhounds",
      MasterEventName: "Races",
      AdvertisedStartTime: "2018-09-20T00:22:09.753Z",
      Venue: {
        Venue: "Daytona Beach",
        StateCode: null,
        Slug: "daytona-beach"
      }
    };

    Date.now = jest.fn().mockReturnValue(1537402868528); //2018-09-20T00:21:08.528Z
    const component = renderer.create(<Event event={event} />);

    expect(component.toJSON()).toMatchSnapshot(); // 1m 1s

    jest.runOnlyPendingTimers()

    expect(component.toJSON()).toMatchSnapshot(); // 1m
  });
});
