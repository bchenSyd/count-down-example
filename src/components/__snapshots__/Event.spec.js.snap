// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render a event tile properly 1`] = `
<div
  className="event"
>
  <div
    className="count-down"
  >
    1m 1s
  </div>
  <div>
    <div
      className="event-name"
    >
      Race 3
    </div>
    <div>
      Daytona Beach
    </div>
  </div>
</div>
`;

exports[`Event should render a event tile properly 2`] = `
<div
  className="event"
>
  <div
    className="count-down"
  >
    1m 
  </div>
  <div>
    <div
      className="event-name"
    >
      Race 3
    </div>
    <div>
      Daytona Beach
    </div>
  </div>
</div>
`;
